src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -51,15 +40,7 @@ const useFetchChannelHook = () => {
     channel.targetName = channelListInfo?.channelName;
     channel.targetImage = channelListInfo?.channelImage;
 
-    if (isAnonymous) {
-      channel.firstMessage = channel?.messages?.[0];
-    } else {
-      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
-      channel.myUserId = signedProfileId;
-      if (channel.firstMessage?.type === 'system') {
-        channel.firstMessage = helperFollowingSystemMessage(channel.firstMessage);
-      }
-    }
+    channel.firstMessage = getFirstMessage(channel?.messages);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This line is an improvement over the previous implementation. It simplifies the process of getting the first message in a channel by using the `getFirstMessage` function from the `useSystemMessage` hook.
```diff
+ channel.firstMessage = getFirstMessage(channel?.messages);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->